MetadataMigration/src/test/java/org/opensearch/migrations/commands/EvaluateTest.java

```diff
@@ -8,10 +8,24 @@
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
 
 class EvaluateTest {
 
+    @Test
+    void evaluate_failsInvalidParameters() {
+        var args = new MigrateArgs();
+        var context = mock(RootMetadataMigrationContext.class);
+        var meta = new MetadataMigration();
+
+        var results = meta.evaluate(args).execute(context);
+
+        assertThat(results.getExitCode(), equalTo(Migrate.INVALID_PARAMETER_CODE));
+        assertThat(results.getErrorMessage(), equalTo("Invalid parameter: No details on the source cluster found, please supply a connection details or a snapshot"));
```

Comment:
I've never looked at the Result class before now.  The class doesn't describe its data in any way - there's no structure to pull important details from it.  I don't see any implementations for getErrorMessage() other than within test code.  Should this pattern be reconsidered?

You should also take a look at CloseableLogSetup.  That will let you test your log outputs without needing to create your own abstractions.  The advantage is that you won't need to carry any test code through production code.

Reply:
getErrorMessage() is used in Evaluate/Migrate. You are right that the clear boundary of the presentation layer components isn't clear - in this case EvaluateResult is a presentation component.

Testing the logger isn't needed for this scenario - its a unit test that ensures the presentation format is prepared in expected way. I've seen CloseableLogSetup, it is used in other tests.